omnigibson/action_primitives/curobo.py

```diff
@@ -0,0 +1,590 @@
+        """
+        # Update obstacles
+        self.update_obstacles()
+
+        # Compute kinematics to get corresponding sphere representation
+        cu_js = lazy.curobo.types.state.JointState(position=self.tensor_args.to_device(q))
+        robot_spheres = self.mg.compute_kinematics(cu_js).robot_spheres
+        # (N_samples, n_obs_spheres, 4) --> (N_samples, 1, n_spheres, 4)
+        robot_spheres = robot_spheres.unsqueeze(dim=1)
+
+        # Run direct collision check
+        with th.no_grad():
+            # Run the overlap check
+            # Sphere shape should be (N_queries, 1, n_obs_spheres, 4), where 4 --> (x,y,z,radius)
+            coll_query_buffer = lazy.curobo.geom.sdf.world.CollisionQueryBuffer()
+            coll_query_buffer.update_buffer_shape(
+                shape=robot_spheres.shape,
+                tensor_args=self.tensor_args,
+                collision_types=self.mg.world_coll_checker.collision_types,
+            )
+
+            dist = self.mg.world_coll_checker.get_sphere_collision(
+                robot_spheres,
+                coll_query_buffer,
+                weight=th.tensor([50000.0], device=self.tensor_args.device),
+                activation_distance=th.tensor([activation_distance], device=self.tensor_args.device),
+                env_query_idx=None,
+                return_loss=False,
+            ).squeeze(
+                dim=1
+            )  # shape (N_samples, n_spheres)
+
+            # Non-zero distances correspond to a collision detection (or close to a collision, within activation_distance
+            # So valid collision-free samples are those where max(n_obs_spheres) == 0 for a given sample
+            collision_results = dist.max(dim=-1).values != 0
+
+        # Return results
+        return collision_results  # shape (B,)
+
+    def compute_trajectory(
+        self,
+        target_pos,
+        target_quat,
+        is_local=False,
+        max_attempts=5,
+        enable_finetune_trajopt=True,
+        return_full_result=False,
+        success_ratio=None,
+        attached_obj=None,
+    ):
+        """
+        Computes the robot joint trajectory to reach the desired @target_pos and @target_quat
+
+        Args:
+            target_pos ((N,3)-tensor): (N, 3) or (3,)-shaped tensor, where N is expected to be self.batch_size, and each
+                entry is (x,y,z) position to reach. If only a single (3,) array is given but N > 1, it will
+                be broadcast to create the appropriate array.
+            target_quat ((N,4)-tensor): (N, 4) or (4,)-shaped tensor, where N is expected to be self.batch_size, and each
+                entry is (x,y,z,w) quaternion orientation to reach. If only a single (3,) array is given but N > 1,
+                it will be broadcast to create the appropriate array.
+            is_local (bool): Whether @target_pos and @target_quat are specified in the robot's local frame or the world
+                global frame
```

Comment:
do we also need to distinguish world frame v.s. scene frame?

Reply:
leaving for TODO for future PR